### PR TITLE
Select autocomplete

### PR DESCRIPTION
**Pull request by djacoby04**
## Summary
Implementation of reusable autocomplete component, which is currently used to select a practitioner for the evaluate measure page. When the user clicks on the autocomplete textbox, a list of up to 10 practitioners display and narrow down based on what the user types. 
## New behavior
On the Evaluate Measure page for a valid measure, an autocomplete component for a practitioner now appears. This enables the user to select an existing practitioner from the dropdown menu, which offers more specific options as the user types. _Note that the screen currently displays which option has been selected beneath the autocomplete component. This is to keep track of the state variable for later use._
## Code changes
- **pages/[resource]/[id]/SelectComponent**  defines an autocomplete textbox that provides a dropdown menu of all the resources of a given resourceType matching what the user types. 
- **pages/[resource]/[id]/evaluate.tsx**  renders autocomplete 
- **__tests__/pages/resource/id/evaluate.test.tsx**, unit tests added to file
-**__tests__/components/SelectComponent.test.tsx** unit tests added to file

## Testing Guidance
See the README.md for first time setup instructions.
#### Testing in browser: 
- To start the frontend: `npm run dev` then navigate to http://localhost:3001 in a browser
- Click on “Measure” from the left-hand-side navigation menu. Click on any of the IDs. Click on the “Evaluate Measure” button near the top right of the main page body.
- To test when no practitioners exist
  - First test the server when no practitioners exist. In this case, the webpage should display "No resources of type Practitioner found"
- To test when a practitioner does exist
  - Start by adding two new practitioners by clicking on the practitioner tab, then clicking the create new Practitioner button, then inputting a valid FHIR practitioner resource, and finally clicking submit. Return to the Evaluate Measure page using the steps above. Then complete the following three checks:
  - click on the textbox and verify that both resources display as part of a drop down menu
  - begin typing out one of the unique practitioner IDs and verify that the drop down menu now only displays the matching practitioner 
  - type something not contained in either practitioner ID to verify that the dropdown menu disappears where there are no matches. 
#### Unit testing: 
- Run the unit tests: `npm run test`

### CODE REVIEW

File: components/SelectComponent.tsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    <Autocomplete
+      value={practitionerValue}
+      onChange={setPractitionerValue}
+      label={`Select ${resourceType}`}
+      placeholder="Start typing to see options"
+      data={myArray}
+      limit={10}
+    />
+  );
+};
```
Comment (elsaperelli) [inline components/SelectComponent.tsx:99]: This is just a suggestion but I am wondering if the code would look better and be a little more concise if these two functions were combined into one? Maybe you tried this and that's why they are separate...just curious!

**Review by elsaperelli — CHANGES_REQUESTED** [review]
Great job! Just a few nit-picky things and one suggestion.

File: components/SelectComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  } else {
+    return <div> {`No resources of type ${props.resourceType} found`} </div>;
+  }
+}
+
+/**
+ * @param fhirJson.BundleEntry
+ * @returns an autocomplete select component populated with resource IDs
+ */
+const PopulateSelect = (
```
Comment (natjoe4) [inline components/SelectComponent.tsx:79]: nitpick, but let's make this a function. Everything else in this file is and seems worth keeping to that standard

File: pages/[resourceType]/[id]/evaluate.tsx [hunk 3]
```diff
@@ -1,15 +1,27 @@
...
   if (resourceType === "Measure") {
-    return <div>Coming soon...</div>;
+    return (
+      <div>
+        <SelectComponent
+          resourceType="Practitioner"
+          setPractitionerValue={setPractitionerValue}
+          practitionerValue={practitionerValue}
+        />
+        <div> {practitionerValue} </div>
```
Comment (natjoe4) [inline pages/[resourceType]/[id]/evaluate.tsx:22]:
```suggestion
        <div> {practitionerValue} </div>
```
I don't think we need this, since the value should be displayed in the autocomplete

File: components/SelectComponent.tsx [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+ * @param props include the string resourceType
+ * @returns a component with a loading component, server error, or autocomplete select component populated with resource IDs
+ */
+export default function SelectComponent(props: {
+  resourceType: string;
+  setPractitionerValue: Dispatch<SetStateAction<string>>;
+  practitionerValue: string;
+}) {
+  const resourceType = props.resourceType;
+  const [pageBody, setPageBody] = useState<fhirJson.Bundle>();
```
Comment (natjoe4) [inline components/SelectComponent.tsx:15]: I think pageBody doesn't quite capture what this variable is storing. `responseBody` may be more appropriate. Also v open to suggestions

File: components/SelectComponent.tsx [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+          console.log(error.message);
+          setFetchingError(true);
+          setLoadingRequest(false);
+        });
+    }
+  }, [resourceType]);
+
+  return loadingRequest ? (
+    <div>Loading content...</div>
+  ) : !fetchingError && pageBody ? (
```
Comment (natjoe4) [inline components/SelectComponent.tsx:40]: never been a huge fan of chaining ternary operators from a readability standpoint. You could just break out these situations into if statements, each with their own return statement inside. Thoughts?

File: __tests__/pages/resource/id/evaluate.test.tsx [hunk 6]
```diff
@@ -1,10 +1,49 @@
...
+import { fhirJson } from "@fhir-typescript/r4-core";
+
+const RESOURCE_ID_BODY: fhirJson.Bundle = {
+  resourceType: "Bundle",
+  meta: {
+    lastUpdated: "2022-06-23T19:52:58.721Z",
+  },
+  type: "searchset",
+  total: 2,
+  entry: [
```
Comment (natjoe4) [inline __tests__/pages/resource/id/evaluate.test.tsx:15]: Here I go getting picky again, but can we change all the places is says diagnostic report here to Practitioner? Just makes things a bit less confusing and more accurate in the fixture.

**Review by natjoe4 — CHANGES_REQUESTED** [review]
Great work! Code works well and I like the reusability of the component. A couple small comments which should be easy to fix. Also, I think the SelectComponent itself could be condensed a bit. I think the workflow of three functions that take largely the same props and need to be chained together to produce a component could make things a bit hard to follow and less modular. Happy to talk about other options if you'd like. That being said, this works great, so if refactoring is a drag, we can push it down the priorities list.

File: components/SelectComponent.tsx (hunk 5 above)
Comment (djacoby04) [reply components/SelectComponent.tsx:50]: Similar response here as with the .then comment--happy to make the change, but a lot of other docs in our code use the same format. If I change this one, should I change those as well?

Comment (natjoe4) [reply components/SelectComponent.tsx]: Lol true. very much a preference thing for me, so I'm happy to just leave it as is

File: components/SelectComponent.tsx [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+}
+
+/**
+ * @param props include the type interface SelectComponentProps
+ * @returns a component with a loading component, server error, or autocomplete select component populated with resource IDs
+ */
+export default function SelectComponent(props: SelectComponentProps) {
+  const resourceType = props.resourceType;
+  const setValue = props.setValue;
+  const value = props.value;
```
Comment (mgramigna) [inline components/SelectComponent.tsx:24]: Coming in hot: if you're gonna do this, might as well do this:

```suggestion
export default function SelectComponent({ resourceType, setValue, value }: SelectComponentProps) {
```

Read more about detructuring here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment

**Review by elsaperelli — APPROVED** [review]
Great work with the changes! lgtm

**Review by natjoe4 — APPROVED** [review]
Looks good to me!
